### ISSUE TITLE
Do items that don't have any interactions during model.fit() affect the model?

### ISSUE THREAD

**Issue by DaStapo** [lyst/lightfm#469]
In my scenario I'm adapting every item's feature depending on the conditions during the prediction. An example of this would be the `'ItemAge:new'` feature which will for certain items change into `'ItemAge:old'` in future. So during `model.fit()`, in addition to the items (n=1000) which are part of the training interactions, I include the same amount of dummy item id's without any features (total n_items = 2000). As I call `model.predict()`, I only predict the dummy items ([1000, 1001 ... 2000]) and I assign the `item_features` paramater with their "current" features. First of all, is there a better approach to this? And secondly, do I understand correctly that during 'model.fit()' the only `item_features` that matter are the ones of items, which are part of the train interactions?

**Comment by 1adarshg** [comment]
I have the same question but for users. 
Since the model cannot predict for users it has not been trained on, if the expected new users are added initially with no interactions , does it affect the model training if I use `model.fit_partial()` on them later to predict ?

**Comment by DaStapo** [comment]
I've run some tests on my custom dataset and I came to the conclusion that the item features of my _dummy_ items during `.fit()` do matter. First I included the item features of items, which had no train interactions, to the `.fit()`'s `item_features`. The predictions were considerably better compared to the second attempt, where the fitting only had the train item's features and the new features were only given during `.predict()`.

@1adarshg 
I've tested splitting up the train data into two and using `.fit_partial()` on the second half and the results were identical, as long as I've included the complete `item_features` of both halves into both rounds of fitting. As soon as i've split up the features into halves aswell, the results went to crap. 

I would close this issue, but there's a chance that I've made a mistake in my tests and maybe someone knows the facts or has diffierent conclusions.
